### ISSUE TITLE
Order results based on usage

### ISSUE THREAD

**Issue by NeighNeighNeigh** [joshbduncan/AiCommandPalette#24]
It would be great if AiCommandPalette was able to learn from usage and order results based on their usage frequency.

For example, let's say I want to align left:
I first enter the keyword 'ali' into the command palette, 
then I choose `Object > Align > Horizontal Align Left` from the menu. 
Now, next time I want to align left, it should be the first item in the menu.

The Command Pallet is learning how I use Illustrator and moving entries to the top of the list based on my usage.

The productivity tool Alfred does this very well, and has an article explaining the implementation [here](https://www.alfredapp.com/help/kb/understanding-result-ordering/) 

Thanks for the great tool!

**Comment by joshbduncan** [comment]
Yeah, I completely agree! I actually have plans to do this at some point. The "Recent Commands" feature I added a bit ago was actually the first building block of this. My original plan was just to more heavily weight recently used commands in the search results, but I had not thought about the "Keyword Latching" concept.

To be honest, my search weighting algorithm (that is a very generous use of the word) is pretty dumb. If you type just "al", `Object > Align > Horizontal Align Left` is at the top, but then if you add the "i" it jumps to the bottom of the results. It weights the result by matched letters of words in a command, with heavier weights for complete/exact word matches. So, in your use case I would type "al l" or another example would be "ar ty" for `Type > Area Type Options...`. This is not ideal and certainly not something I explain in the README, so it needs to be improved.

Not sure when I'll get to it as I have a newborn baby girl and time is very limited at the moment, but I will certainly let you know when I do.

Funny thing, I'm a heavy Alfred user (for many years) and I did not even realize the "Keyword Latching" feature at work. I actually created this script because I wanted the Alfred functionality within Illustrator.

Cheers!

**Comment by NeighNeighNeigh** [comment]
Hey thanks for getting back to me :) Sounds like we are on the same page!
I really appreciate your extension and look forward to seeing where it goes, patiently of course, while wishing you all the best with the new little one.

**Comment by joshbduncan** [comment]
So, I've got a big release coming soon (v.0.10.0) with lots of changes. I still have two things to do before I release it, build a start-up screen customizer/builder, and implement the better searching/filtering algo you suggested.

As I mentioned above, I already had some code that would give a higher weight/ranking/score to commands that were used recently but there was a bug in the code so it wasn't working. I have since fixed that in the [v.0.10.0 branch](https://github.com/joshbduncan/AiCommandPalette/commit/ce4c6070f0acb24fc5bb851dcd50a4ec5e8c14e6) it helps quite a bit getting back to recent commands faster.

I'll post an update here when I've got more ready. Thanks again!
